### ISSUE TITLE
Speed up TDD/BDD workflow by including a watcher

### ISSUE THREAD

**Issue by matcarey** [gruntjs/grunt-contrib-jasmine#202]
TL;DR: If grunt-contrib-jasmine included a filewatcher then we could keep a PhantomJS instance available and test faster than we currently can using a mix of grunt-contrib-watch and grunt-contrib-jasmine.

When working on a TDD/BDD project with Jasmine I usually setup a gruntflle which watches and runs the tests on every save.  Any delay between the save and the end of the unit tests breaks the TDD flow and has a massive impact on the way developers interact with the tests.  The codebase I'm using as an example ([js-config](https://github.com/dorightdigital/js-config)) has 48 specs which run in 0.059s however running them in Grunt takes about 1.75 seconds.  This time of course includes some Grunt setup etc. but I'd estimate (based on the figures below) that about 1 second is grunt-contrib-jasmine starting up and creating a phantom instance.

I have a codebase which is designed for node and the browser.  To make sure I'm covering both cases I run the tests with grunt-contrib-jasmine and grunt-jasmine-nodejs - each of these run the exact same test from the same gruntfile.  I've pushed a little timer script to https://github.com/dorightdigital/js-config/blob/jasmine-performance/jasmine_performance.rb which on my machine outputs:

````
`grunt jasmine` took 1.783 seconds.  Jasmine's time was 0.027 seconds.
`grunt jasmine` took 1.776 seconds.  Jasmine's time was 0.026 seconds.
`grunt jasmine` took 1.777 seconds.  Jasmine's time was 0.026 seconds.
`grunt jasmine` took 1.778 seconds.  Jasmine's time was 0.028 seconds.
`grunt jasmine` took 1.788 seconds.  Jasmine's time was 0.026 seconds.
`grunt jasmine` took 1.773 seconds.  Jasmine's time was 0.024 seconds.
`grunt jasmine` took 1.782 seconds.  Jasmine's time was 0.029 seconds.
`grunt jasmine` took 1.91 seconds.  Jasmine's time was 0.159 seconds.
`grunt jasmine` took 1.776 seconds.  Jasmine's time was 0.027 seconds.
`grunt jasmine` took 1.774 seconds.  Jasmine's time was 0.029 seconds.
Average time for excluding unit tests was 1.752 seconds.
`grunt jasmine_nodejs` took 0.69 seconds.  Jasmine's time was 0.041 seconds.
`grunt jasmine_nodejs` took 0.696 seconds.  Jasmine's time was 0.045 seconds.
`grunt jasmine_nodejs` took 0.696 seconds.  Jasmine's time was 0.046 seconds.
`grunt jasmine_nodejs` took 0.695 seconds.  Jasmine's time was 0.043 seconds.
`grunt jasmine_nodejs` took 0.695 seconds.  Jasmine's time was 0.046 seconds.
`grunt jasmine_nodejs` took 0.692 seconds.  Jasmine's time was 0.044 seconds.
`grunt jasmine_nodejs` took 0.692 seconds.  Jasmine's time was 0.045 seconds.
`grunt jasmine_nodejs` took 0.682 seconds.  Jasmine's time was 0.043 seconds.
`grunt jasmine_nodejs` took 0.692 seconds.  Jasmine's time was 0.045 seconds.
`grunt jasmine_nodejs` took 0.69 seconds.  Jasmine's time was 0.046 seconds.
Average time for excluding unit tests was 0.648 seconds.
````

My hypothosis is that the delay between saving a file and seeing the output of the tests could be cut by around 1 second if we internalise the watching behaviour so that we can share a phantom instance (and maybe other setup) between runs rather than having to start from scratch every time.

I'm happy to do some dev work to prove this but I wanted to check:
1. Have I missed anything obvious like some configuration which already exists?
2. Has it been tried before and failed?
3. Is there a general appetite for it?

I would imagine adding a watch option to the configuration, if this is set we watch the files which are referenced, if not we exit as normal.

Looking forward to some responses :)
